sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -760,6 +771,18 @@ private static void validateOutputComponent(String partialFilePattern) {
         + partialFilePattern);
   }
 
+  private static String getBrowseUrl(String filePattern) {
+    IOChannelFactory factory;
+    try {
+      factory = IOChannelUtils.getFactory(filePattern);
```

Comment:
My impression of the current code is that the `IOChannelFactory` is never interrogated unless `validate` is true. This should probably stick to that discipline.